### PR TITLE
Try adding focus outline for blocks that don't have input fields

### LINKED ISSUE #1829: Image block keyboard focus and tab order
Tested on: Gutenberg 0.4.

Setup: Mac OS X Sierra, MAMP, Safari 10.1.1.
Gutenberg is the only plugin active.

When you add an image block with the keyboard, it's not clear where the focus goes. VoiceOver says "You are currently on a group".

voiceover add image block

Tabbing through goes from the up/down arrows to cog, delete and then the image positioning and link buttons. Finally, you get to the Insert from Media Library button, after about 12 clicks.

add image gutenberg keyboard focus

Would it not be better to set focus to Insert from Media Library first? The workflow for adding an image in the regular post editor opens up the Insert Media dialog from the Add Media button. I would imagine that most users would want to add an image first before positioning it or moving the block.

### LINKED ISSUE #3560: Consider unfading outlines when moving arrowkeys between blocks
When a non text block has focus, you can press the delete button to delete it. This works really well with the mouse:

delete with mouse

This also works with the keyboard, but not at all as well:

delete with keys

Simply put, there is no indication that the block has focus, because the "selected block outlines" have faded out because we're using arrowkeys.

Some ideas for improvement:

- Unfade outlines for a block as soon as you arrow-key into it. If it's a text block and you continue to arrow through it, the outlines will fade out again, but if it's not a text block the outlines will stay there
- Have a different focus style for non text blocks
- "Blink" the outlines whenever they're supposed to be faded out and you arrowkey between blocks

Thoughts?

### PR DESCRIPTION
**Pull request by jasmussen**
This probably fixes #3560 and maybe fixes #1829.

When you select a block, we show outlines for that block. When you then start writing, those outlines and all the controls fade out, letting you focus on the text. But when you then use arrowkey navigation to move between blocks and encounter a block that has no textfield, there's no indication what's selected. This is nicely illustrated in #3560.

This PR tries to address that, by showing an indicator on blocks that have no input fields. And so in absence of a caret cursor to indicate your place, there's now an outline:

block-indicator

In addition to highlighting your place in the block list, this is also useful in indicating that you can _press Enter_ to make a linebreak, or _press Delete_ or _Backspace_ to delete the indicated block.

### CODE REVIEW

**Review by youknowriad — APPROVED** [review]
I personally like this change, might be good to have a second opinion.

File: editor/edit-post/modes/visual-editor/style.scss [hunk 1]
```diff
@@ -39,6 +39,14 @@
 	}
 }
 
+// This is a focus style shown for blocks that need an indicator even when in an isEditing state
+// like for example an image block that receives arrowkey focus.
+.editor-visual-editor .editor-block-list__block:not( .is-selected ) .editor-block-list__block-edit:focus  {
```
Comment (youknowriad) [inline editor/edit-post/modes/visual-editor/style.scss]: Shouldn't this be moved to the `editor/components/block-list/style.scss` file? Thinking this style could be usefulll regardless of the layout.

**Comment by jasmussen** [comment]
I pushed a tiny tweak to the indicator style:
caret position

**Review by karmatosed — APPROVED** [review]

**Comment by jasmussen** [comment]
🎉

**Pull request merged**